### PR TITLE
listen: Only error out if not able to bind any interface

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Quit only when the server does not find any interface to listen on.

Otherwise print logs why it is not able to bind to other interfaces.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -600,7 +600,8 @@ func serverMain(ctx *cli.Context) {
 	}
 
 	bootstrapTrace("xhttp.NewServer")
-	httpServer := xhttp.NewServer(getServerListenAddrs()).
+	listenAddrs := getServerListenAddrs()
```
Comment (harshavardhana) [inline cmd/server-main.go]: If you want current listenAddrs() move this as a function function in `httpServer.ListenAddrs()`  @vadmeste

File: cmd/server-main.go [hunk 2]
```diff
@@ -612,7 +613,18 @@ func serverMain(ctx *cli.Context) {
...
+			globalHTTPServerErrorCh <- err
+			return
+		}
+		for i, e := range listenErrs {
+			if e == nil {
+				continue
+			}
+			logger.LogIf(GlobalContext, fmt.Errorf("Unable to listen on `%s`: %v", listenAddrs[i], listenErrs[i]))
+		}
+		globalHTTPServerErrorCh <- serveFn()
```
Comment (harshavardhana) [inline cmd/server-main.go]: Pass a `function()` to print the errors if any like this

```
    serve, err := httpServer.Init(GlobalContext, func(listenAddr string, err error) {
    if err != nil {
         logger.LogIf(GlobalContext, fmt.Errorf("Unable to listen on `%s`: %v",  listenAddr, err)
    })
    if err != nil {
        globalHTTPServerErrorCh <- err
	return
    }
    globalHTTPServerErrorCh <- serve()
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/server-main.go (hunk 2 above)
Comment (vadmeste) [reply cmd/server-main.go]: Done

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**